### DISCUSSION TITLE
Vending Xcode schemes

### DISCUSSION THREAD

**Discussion by kamcma** [pointfreeco/swift-composable-architecture#1371] [General]
It's not a huge issue, but I have occasionally been puzzled by schemes from the TCA package showing up in my schemes list in dependent projects.

I can't pretend to be an Xcode expert by any means, but I did find this PR in the `swift-collections` package to be clarifying: apple/swift-collections#155.

Do you think this is the same thing happening with TCA? Is TCA vending these schemes deliberately? Anyway, just wanted to share in case this was a bit of deferred maintenance. If there's a reason for it, however, I'd understand, and carry on.

**Comment by acosmicflamingo** [comment]
OMG yea I have found that to be really annoying too. I've done every single thing I could in Tuist configurations but to no avail. At this point I'm in the acceptance phase but I'd definitely be happy to see those other schemes disappear :D

**Comment by stephencelis** [comment]
@kamcma Can you specify which schemes in particular? We currently have some checked in that were needed for the Swift Package Index to detect whether or not TCA built for watchOS, but perhaps we can git ignore some of the others? Just want to be sure of what you're seeing, exactly, to understand the problem.

**Reply by kamcma** [reply]
Sorry, should have occurred to me to share those specifics up front. Here's what I'm getting in one project, with my schemes redacted:
<img width="848" alt="Screen Shot 2022-09-10 at 4 45 00 p m" src="https://user-images.githubusercontent.com/5169528/189501258-871ea2c9-a85d-4ccf-beb8-e4598531677d.png">

You can delete them of course, but they do eventually return, by what action exactly I am not sure.

**Reply by kamcma** [reply]
Unchecking "Show" likewise cleans up the scheme list, but it does not dirty my working tree, so I suspect it would not persist through a fresh clone.

**Reply by stephencelis** [reply]
We just thought this was an artifact of SPM! We'll prune these down for next version. Sorry about that and thanks for letting us know!

**Reply by acosmicflamingo** [reply]
YEAHHHHHHH!!! :D

**Reply by stephencelis** [reply]
With 0.40.2 (and downstream updates), these schemes are no longer along for the ride. You may need to do a fresh checkout to notice, though! Thanks again for the heads-up!

**Comment by acosmicflamingo** [comment]
<img width="319" alt="image" src="https://user-images.githubusercontent.com/67525430/189500878-92ce2ea1-40b7-440b-9b68-dec78634c060.png">
This is what I'm seeing.

**Comment by acosmicflamingo** [comment]
@stephencelis thank you SO much for merging this! What a nice surprise; I wasn't expecting this to be addressed for a while, with all the other conversations going on in parallel. My schemes list is so clean now :)
